simulatingrisk/hawkdovemulti/batch_run.py

```diff
@@ -0,0 +1,182 @@
+
+from mesa.batchrunner import _make_model_kwargs, _collect_data
+
+from simulatingrisk.hawkdovemulti.model import HawkDoveMultipleRiskModel
+
+
+neighborhood_sizes = list(HawkDoveMultipleRiskModel.neighborhood_sizes)
+
+# combination of parameters we want to run
+params = {
+    "grid_size": [10, 25, 50],  # 100],
+    "risk_adjustment": ["adopt", "average"],
+    "play_neighborhood": neighborhood_sizes,
+    "observed_neighborhood": neighborhood_sizes,
+    "adjust_neighborhood": neighborhood_sizes,
+    "hawk_odds": [0.5, 0.25, 0.75],
+    "adjust_every": [2, 10, 20],
+    "risk_distribution": HawkDoveMultipleRiskModel.risk_distribution_options,
+    "adjust_payoff": HawkDoveMultipleRiskModel.supported_adjust_payoffs,
+    # random?
+}
+
+
+# method for multiproc running model with a set of params
+def run_hawkdovemulti_model(args):
+    run_id, iteration, params, max_steps = args
+    # simplified model runner adapted from mesa batch run code
+
+    model = HawkDoveMultipleRiskModel(**params)
+    while model.running and model.schedule.steps <= max_steps:
+        model.step()
+
+    # collect data for the last step
+    step = model.schedule.steps - 1
+
+    model_data, all_agents_data = _collect_data(model, step)
+
+    # combine run id, step, and params, with collected model data
+    run_data = {"RunId": run_id, "iteration": iteration, "Step": step}
+    run_data.update(params)
+    run_data.update(model_data)
+
+    agent_data = [
+        {
+            "RunId": run_id,
+            "iteration": iteration,
+            "Step": step,
+            **agent_data,
+        }
+        for agent_data in all_agents_data
+    ]
+
+    return run_data, agent_data
+
+
+def batch_run(
+    params, iterations, number_processes, max_steps, progressbar, file_prefix=""
+):
+    param_combinations = _make_model_kwargs(params)
+    total_param_combinations = len(param_combinations)
+    total_runs = total_param_combinations * iterations
+    print(
+        f"{total_param_combinations} parameter combinations, "
+        + f"{iterations} iteration{'s' if iterations != 1 else ''}, "
+        + f"{total_runs} total runs"
+    )
+
+    # create a list of all the parameters to run, with run id and iteration
+    runs_list = []
+    run_id = 0
+    for params in param_combinations:
+        for iteration in range(iterations):
+            runs_list.append((run_id, iteration, params, max_steps))
+            run_id += 1
+
+    # collect data in a directory for this model
+    data_dir = os.path.join("data", "hawkdovemulti")
+    os.makedirs(data_dir, exist_ok=True)
+    datestr = datetime.today().isoformat().replace(".", "_").replace(":", "")
+    model_output_filename = os.path.join(data_dir, f"{file_prefix}{datestr}_model.csv")
+    agent_output_filename = os.path.join(data_dir, f"{file_prefix}{datestr}_agent.csv")
+    print(
+        "Saving data collection results to:\n  %s\n  %s"
+        % (model_output_filename, agent_output_filename)
```

Comment:
Switch to using f-strings
```suggestion
        f"Saving data collection results to:\n  {model_output_filename}"
        + f"\n  {agent_output_filename}"
```